### PR TITLE
[indirect-value] add new port

### PR DESCRIPTION
**Pull request by jiayuehua**
- [X] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [X] The name of the port matches an existing name for this component on https://repology.org/ if possible, and/or is strongly associated with that component on search engines.
- [X] Optional dependencies are resolved in exactly one way. For example, if the component is built with CMake, all `find_package` calls are REQUIRED, are satisfied by `vcpkg.json`'s declared dependencies, or disabled with [CMAKE_DISABLE_FIND_PACKAGE_Xxx](https://cmake.org/cmake/help/latest/variable/CMAKE_DISABLE_FIND_PACKAGE_PackageName.html)
- [X] The versioning scheme in `vcpkg.json` matches what upstream says.
- [X] The license declaration in `vcpkg.json` matches what upstream says.
- [X] The installed as the "copyright" file matches what upstream says.
- [X] The source code of the component installed comes from an authoritative source.
- [X] The generated "usage text" is accurate. See [adding-usage](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/examples/adding-usage.md) for context.
- [X] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [X] Only one version is in the new port's versions file.
- [X] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/indirect-value/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+vcpkg_cmake_configure(
+    SOURCE_PATH "${SOURCE_PATH}"
+    OPTIONS 
+        -DENABLE_CODE_COVERAGE=OFF
+        -DENABLE_INCLUDE_NATVIS=OFF
+        -DBUILD_TESTING=OFF
+)
+
+vcpkg_cmake_install()
+vcpkg_cmake_config_fixup(PACKAGE_NAME indirect-value CONFIG_PATH lib/cmake/indirect_value)
```
[the remark this answers is not in the document]
Comment (dg0yt) [reply ports/indirect-value/portfile.cmake:22]: I don't ask for changing the *vcpkg* port name. It is about the *CMake* package name as defined upstream. Actually the `PACKAGE_NAME?  specifies the directory name where the CMake config will be stored. This must match the CMake package name:

If the config file is named `indirect_value-config.cmake` or `indirect_valueConfig.cmake`, then the CMake package name is `indirect_config`, and the directory must match `indirect_config*`, case-insensitively. (`indirect_config-0.8.15` is valid, `indirect-config` isn't. )

So yes, we get two directories in `share`.

[the remark this answers is not in the document]
Comment (jiayuehua) [reply ports/indirect-value/portfile.cmake:22]: your suggestion is change it to underscore? even may result that get two dir in share.

[the remark this answers is not in the document]
Comment (dg0yt) [reply ports/indirect-value/portfile.cmake:22]: We must not change upstream's choice of CMake package name.

File: ports/indirect-value/fix-install.patch [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+         INSTALL_DESTINATION
+             "${CMAKE_INSTALL_LIBDIR}/cmake/indirect_value"
+     )
+@@ -199,7 +194,7 @@ if (${CPP_INDIRECT_IS_NOT_SUBPROJECT})
+ 
+     install(
+         FILES
+-            ${CMAKE_CURRENT_BINARY_DIR}/indirect_value-config.cmake
++            ${CMAKE_CURRENT_BINARY_DIR}/indirect-value-config.cmake
+             ${CMAKE_CURRENT_BINARY_DIR}/indirect_value-version.cmake
```
Comment (dg0yt) [inline ports/indirect-value/fix-install.patch:40]: These changes have to be removed. Note that version file no longer match the package names...

File: ports/indirect-value/portfile.cmake (hunk 1 above)
[the remark this answers is not in the document]
Comment (jiayuehua) [reply ports/indirect-value/portfile.cmake:22]: ok, I will revert it back.

**Comment by MonicaLiu0311** [comment]
The usage test passed (header files found):
```
The package indirect-value provides CMake targets:

    find_package(indirect_value CONFIG REQUIRED)
    target_link_libraries(main PRIVATE indirect_value::indirect_value)
```

**Review by MonicaLiu0311 — APPROVED** [review]

**Pull request merged**